biz.aQute.bndlib/src/aQute/bnd/build/model/BndEditModel.java

```diff
@@ -967,14 +969,117 @@ public void setTestSuites(List<String> suites) {
 	}
 
 	public List<HeaderClause> getPlugins() {
-		return doGetObject(Constants.PLUGIN, headerClauseListConverter);
+		// return all plugins
+		// we do prefix matching to support merged properties like
+		// -plugin.1.Test, -plugin.2.Maven etc.
+		try {
+			Processor proc = getProperties();
+			Properties allProps = proc.getProperties();
+			Set<String> propertyKeys = proc.getPropertyKeys(true);
+
+			List<HeaderClause> headers = propertyKeys.stream()
+				.filter(p -> p.startsWith(Constants.PLUGIN))
+				.map(p -> headerClauseListConverter.convert(allProps.getProperty(p)))
+				.flatMap(List::stream)
+				.toList();
+
+			return headers;
+
+		} catch (Exception e) {
+			throw Exceptions.duck(e);
+		}
+
 	}
 
 	public void setPlugins(List<HeaderClause> plugins) {
 		List<HeaderClause> old = getPlugins();
 		doSetObject(Constants.PLUGIN, old, plugins, complexHeaderClauseListFormatter);
 	}
 
+	/**
+	 * Similar to {@link #getPlugins()} but returns a map where the key is the
+	 * property key of the bnd file e.g.
+	 * <code>-plugin.1.Test, -plugin.2.Maven </code> The value is a List of
+	 * plugins, although usually it is just a 1-element list. But it is also
+	 * possible to specify multiple plugins under a single key, thus it is a
+	 * list.
+	 *
+	 * @return a map with a property keys and their plugins.
+	 */
+	public Map<String, List<MergedHeaderClause>> getPluginsProperties() {
+		// return all plugins
```

Comment:
I think this logic should be moved to Processor, next to `mergeProperties` so it is easier to track when we make changes in one or the others. Since this is processor, I think we need a call like:

    Processor.getMergedParts(String keyStem) : MergedPart(Processor p, String keyStem, String actualKey, String value)

We could even rewrite ( one day) the mergedProperties with this method. In the BndEditModel, you then turn each one into a number of constituent clauses that can be edited if the processor == the BndEditModel's processor. This would be similar to your MergedHeaderClause, but I would make it something like:

     MergedHeaderClause( MergedPart part, HeaderClause), `isLocal()` can check the part if it is the same Processor.